.github/workflows/ci.yml

```diff
@@ -13,50 +13,103 @@ on:
 env:
   BUILD_TYPE: Release
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   build:
     runs-on: ${{ matrix.os }}
+
+    continue-on-error: true
+
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-latest, macOS-latest]
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        python-version:
+          - "3.8"
+          - "3.11"
+
+    defaults:
+      run:
+        shell: bash -l {0}
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
 
-    - name: Install gcc on macos
-      if: contains( matrix.os, 'macos')
-      run: brew install gcc@9 || brew upgrade gcc@9 || true
+    # this will set the system compiler;
+    # This must be done *before* setting up miniconda, see:
+    # https://github.com/ilammy/msvc-dev-cmd/issues/34
+    - name: Set Windows env
+      if: matrix.os == 'windows-latest'
+      uses: ilammy/msvc-dev-cmd@v1
 
-    - name: Set up Python 3.7
-      uses: actions/setup-python@v2
+    - name: Cache conda
+      uses: actions/cache@v3
+      env:
+        # Increase this value to reset cache if etc/example-environment.yml has not changed
+        CACHE_NUMBER: 0
       with:
-        python-version: 3.7
-
-    - name: Install Python dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install pytest numpy h5py scipy pandas scikit-build pybind11
-        pip install git+https://gitlab.com/reinholdt/polarizationsolver.git@master
+        path: ~/conda_pkgs_dir
+        key:
+          ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('.github/environment.yml') }}
 
-    - name: Configure CMake (Linux)
-      if: contains( matrix.os, 'ubuntu')
-      run: cmake -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DPYTHON_EXECUTABLE=$(which python) -DENABLE_PYTHON_INTERFACE=ON
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        auto-update-conda: true
+        auto-activate-base: false
+        activate-environment: cppe-gha
+        environment-file: .github/environment.yml
```

Comment:
I tried mamba actually, but for some reason it failed to solve 🤔

Reply:
huh, gtk

Reply:
But i think it's because of my insistence on tar.bz2. I'll bust the cache and retry without that option. Let's see. Maybe I get the caching cake and I eat it too with mamba